jina/peapods/networking.py

```diff
@@ -451,6 +451,50 @@ async def task_wrapper(requests, stubs):
 
         return asyncio.create_task(task_wrapper(requests, connection))
 
+    @staticmethod
+    def get_grpc_channel(address: str, options: Optional[list] = None) -> grpc.Channel:
+        """
+        Creates a grpc channel to the given address
+
+        :param address: The address to connect to, format is <host>:<port>
+        :param options: A list of options to pass to the grpc channel
+
+        :return: A grpc channel
+        """
+        if options == None:
+            options = GrpcConnectionPool.get_default_grpc_options()
+
+        port = address.split(':')[-1]
+
+        if port == '443':
+            return grpc.secure_channel(address, grpc.ssl_channel_credentials(), options)
+
+        return grpc.insecure_channel(address, options)
+
+    @staticmethod
+    def get_grpc_aio_channel(
+        address: str, options: Optional[list] = None
+    ) -> grpc.aio.Channel:
+        """
+        Creates a async grpc channel to the given address
+
+        :param address: The address to connect to, format is <host>:<port>
+        :param options: A list of options to pass to the grpc channel
+
+        :return: A async grpc channel
+        """
+        if options == None:
+            options = GrpcConnectionPool.get_default_grpc_options()
+
+        port = address.split(':')[-1]
+
+        if port == '443':
```

Comment:
what is this?

Reply:
Using SSL channel if the port is 443, which is dedicated port for HTTPS.

Reply:
it should be an option and not a hack like this I would say

Reply:
addressed